### PR TITLE
Generate Managers and POMs for Azure SDK for Java

### PR DESCRIPTION
**Pull request by RikkiGibson**
Adds some templates to generate stub Manager classes. The generation of the Manager and POM is controlled by the RegenerateManager flag. Note that the path provided to Autorest in the Azure SDK for Java project has changed so that we can reasonably write the POM to the correct location.

The ServiceName property is something that we just expect to work most of the time due to the convention for paths in Azure, as well as the fact that we expect each service we generate code for to have at least one method.

### CODE REVIEW

**Comment by jianghaolu** [comment]
@azuresdkci add to whitelist

File: src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+}
+<!--
+@Header(" * ").TrimMultilineHeader()
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.microsoft.azure</groupId>
+        <artifactId>azure-parent</artifactId>
+        <version>1.1.3-SNAPSHOT</version>
```
Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: Do you think this version can go to the code generator? Same as the Beta annotation in the manager template.

Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: You can get the beta annotation version from this version though - 1.x.0-SNAPSHOT can generate @Beta(SinceVersion=V1_X_0) and 1.x.y-SNAPSHOT can generate @Beta(SinceVersion=V1_{X+1}_0)

**Comment by jianghaolu** [comment]
Seems like you need to run `gulp regenerate` to reflect the changes for the tests.

File: src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml (hunk 1 above)
Comment (RikkiGibson) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: Would we pass a value from CodeGeneratorJvaf to the template indicating which version of Azure SDK for Java is being targeted?

Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: Yes - that would be ideal

Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: One day we can do a cmd argument but I think we can do that later

Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: The idea is that the generated manager class will be able to be merged into master without further editing.

**Comment by RikkiGibson** [comment]
Will need to troubleshoot my gulp and node setup to get `gulp regenerate` to run without exploding.

File: src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml (hunk 1 above)
Comment (RikkiGibson) [inline src/generator/AutoRest.Java.Azure/Templates/AzurePomTemplate.cshtml]: What's the input/output you'd like to see? Someone passes `-AzureSDKVersion 1.1.3`, then the pom gets `1.1.3-SNAPSHOT` and the SinceVersion is `V1_2_0`?

**Comment by fearthecowboy** [comment]
Did you install your workstation by running: https://raw.githubusercontent.com/Azure/autorest/master/Tools/setup-developerworkstation.ps1 as admin?

That should have set everything up to make gulp regenerate work.

G

**Comment by RikkiGibson** [comment]
Thank you. I took the steps you both mentioned and now I seem to be regenerating fine.

**Comment by RikkiGibson** [comment]
@jianghaolu Seems like about 880 files were changed through regeneration--should I be committing all of that? Or just picking out Java.Azure.Fluent related stuff?

**Comment by fearthecowboy** [comment]
What did you change!?

(if it’s just CRLF changes, no worries)

**Comment by RikkiGibson** [comment]
I'm seeing a lot of:

```
- // Code generated by Microsoft (R) AutoRest Code Generator 1.1.0.0
+ // Code generated by Microsoft (R) AutoRest Code Generator 1.2.0.0
```

**Comment by fearthecowboy** [comment]
Pull from master and rebuild/regenerate – those should be updated already.

G

File: src/generator/AutoRest.Java.Azure/Model/CodeModelJva.cs [hunk 2]
```diff
@@ -80,5 +80,34 @@ public string ServiceName
...
+        /// </summary>
+        [JsonIgnore]
+        public string BetaSinceVersion
+        {
+            get
+            {
+                var versionParts = targetVersion.Split('.');
+                var minorVersion = int.Parse(versionParts[1]) + 1;
+                var result = "V" + versionParts[0] + "_" + minorVersion + "_0";
+                return result;
```
Comment (jianghaolu) [inline src/generator/AutoRest.Java.Azure/Model/CodeModelJva.cs]: one more nit picking - 2.0.0-SNAPSHOT will give me V2_1_0 but it should be V2_0_0 - you need to check the last digit to see if it's `0`.

**Comment by RikkiGibson** [comment]
I merged in master and regenerated again. Now I'm here:
```
- // Code generated by Microsoft (R) AutoRest Code Generator 1.2.1.0
+ // Code generated by Microsoft (R) AutoRest Code Generator 1.2.2.0
```

**Comment by fearthecowboy** [comment]
Well, that's more like it -- although not unexpected.  I guess we forgot to regenerated when we bumped the version number.  

Go ahead and add those changes into your PR.

**Comment by RikkiGibson** [comment]
Did you want me to open a new PR, @fearthecowboy?

File: src/generator/AutoRest.Java.Azure/Model/CodeModelJva.cs (hunk 2 above)
Comment (RikkiGibson) [inline src/generator/AutoRest.Java.Azure/Model/CodeModelJva.cs]: Thanks for catching that. Should there be unit tests for this? Or we'll just run it and see if it did the right thing?

**Comment by fearthecowboy** [comment]
Whoops. hit wrong button.

**Comment by jianghaolu** [comment]
@RikkiGibson I think there's still quite some diff - http://azuresdkci.cloudapp.net/job/autorest/5692/console

**Comment by RikkiGibson** [comment]
@jianghaolu I'm not sure how to interpret the output from CI. Did I get something wrong when regenerating sources?

**Comment by fearthecowboy** [comment]
@olydis -- we're seeing this variation again in the CI with generating source maps. 

O_o

**Comment by olydis** [comment]
Seems to depend on weird details... maybe it's non-deterministic. Let's just not compare source maps in CI for now, even though there are differences they don't seem crucial.

**Comment by RikkiGibson** [comment]
Should I be doing anything to make that happen? Or will you guys go in and tweak that after this PR?

**Comment by fearthecowboy** [comment]
He's fixing the issue in the 2.0 branch -- You're going to need to rebase and issue a new PR there, but *after* that gets committed.

**Pull request closed without merging**